ksqldb-functional-tests/src/test/resources/query-validation-tests/group-by.json

```diff
@@ -572,6 +572,83 @@
         ]
       }
     },
+    {
+      "name": "multiple expressions - KAFKA to struct key format",
```

Comment:
unless we already have some, can we add negative tests of changing from `JSON` to `KAFKA` in a way that doesn't work? and similarly, trying to have delimited with multi-columns where they are no all primitives

Reply:
+1 to the additional tests @agavra suggested. Let's also add one for converting from KAFKA key format to an unsupported primitive type such as DECIMAL (similar to the relevant unit test you've already added above).